### PR TITLE
ldc 1.30.0

### PR DESCRIPTION
**Pull request by Porkepix**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by carlocab** [comment]
```
  /home/linuxbrew/.linuxbrew/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/llvm/14.0.6_1/lib/libLLVM-14.so: undefined reference to `std::__exception_ptr::exception_ptr::_M_addref()@CXXABI_1.3.13'
  clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by cho-m** [comment]
> /home/linuxbrew/.linuxbrew/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/llvm/14.0.6_1/lib/libLLVM-14.so: undefined reference to `std::__exception_ptr::exception_ptr::_M_addref()@CXXABI_1.3.13'
>   clang-14: error: linker command failed with exit code 1 (use -v to see invocation)

We also saw this in previous attempt when `llvm` was on LLVM 13 (https://github.com/Homebrew/homebrew-core/pull/99496#issuecomment-1101036524). May be result of building `llvm` with `gcc-11`.

Can't say about impact, but can try using `-stdlib=libc++`.

---

Also, can remove the `uses_from_macos "llvm" ...` workaround which is due to `llvm@12` not being detected by compiler selection (`llvm` should be)

**Comment by danielnachun** [comment]
I think we're not even adding the necessary `CMAKE_*_LINKER_FLAGS` that are always needed when building C++ with `clang` on Linux.  I'll have to do a local test but I suspect it's just a matter of using right headers and shared libraries like we do in the `llvm` formula.

**Comment by danielnachun** [comment]
I pushed a fix that work locally for me.  This flags should eventually handled by our compiler shims, but it doesn't make sense to do so until we've started using `lib/gcc/current` for the `gcc` formula (and we've migrated to GCC 12 as `gcc`).  Note that on Linux, any time C++ packages are build with LLVM they **must** use the newest version of `libstdc++` to make sure they support the C++ standards that the current version of LLVM.  It's pretty rare that we do this right now so it won't add too many long term dependencies to `gcc`, unless we eventually use brewed GCC to build everything at which point it would be seamless.

File: Formula/ldc.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Ldc < Formula
   depends_on "cmake" => :build
   depends_on "libconfig" => :build
   depends_on "pkg-config" => :build
-  depends_on "llvm@12"
+  depends_on "llvm"
 
   uses_from_macos "libxml2" => :build
   # CompilerSelectionError: ldc cannot be built with any available compilers.
```
Comment (cho-m) [inline Formula/ldc.rb:29]: Should hopefully be able to remove this and the following line now if we no longer rely on `llvm@12` (which Homebrew doesn't consider a valid compiler for selection)

Comment (danielnachun) [reply Formula/ldc.rb]: I've dropped those lines and everything worked great, so they are no longer needed.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**